engine/src/main/java/com/datatorrent/stram/stream/InlineStream.java

```diff
@@ -80,14 +88,29 @@ public void teardown()
   public void put(Object tuple)
   {
     try {
-      super.put(tuple);
+      reservoir.put(tuple);
+      if (!(tuple instanceof Tuple)) {
```

Comment:
Please see http://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java.